Cauli/CauliViewController/CauliViewController.swift

```diff
@@ -24,14 +24,25 @@ import UIKit
 
 internal class CauliViewController: UITableViewController {
 
-    typealias DisplayableFloret = Floret & Displayable
-
     private let cauli: Cauli
     private var displayableFlorets: [DisplayableFloret]
+    private var interceptableFlorets: [InterceptableFloret]
 
     init(cauli: Cauli) {
         self.cauli = cauli
-        displayableFlorets = cauli.florets.compactMap { $0 as? (DisplayableFloret) }
+        
+        var displayableFlorets: [DisplayableFloret] = []
+        var interceptableFlorets: [InterceptableFloret] = []
+        cauli.florets.forEach {
+            switch $0 {
+            case let displayableFloret as DisplayableFloret: displayableFlorets.append(displayableFloret)
+            case let interceptableFloret as InterceptableFloret: interceptableFlorets.append(interceptableFloret)
+            default: assertionFailure("useful message goes here")
+            }
+        }
```

Comment:
I am not sure if this is actually better, but a different possibility would be:

```swift
var (displayableFlorets, interceptableFlorets) = cauli.florets.reduce(([],[])) { (state, floret) in
    var displayableFlorets = state.0
    var interceptableFlorets = state.1
    // add the florets to the one or the other, or both
    return (displayableFlorets, interceptableFlorets)
}
```

Reply:
I decided to go with a much more readable solution and I'm now iterating twice through the florets.

Reply:
Yeah, I like it much more now!